src/test/java/org/commcare/formplayer/services/FormSessionsPaginationTest.java

```diff
@@ -0,0 +1,80 @@
+package org.commcare.formplayer.services;
+
+import com.google.common.collect.ImmutableMap;
+
+import org.commcare.formplayer.beans.FormsSessionsRequestBean;
+import org.commcare.formplayer.objects.FormSessionListView;
+import org.commcare.formplayer.objects.SerializableFormSession;
+import org.commcare.formplayer.repo.FormSessionRepo;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.cache.CacheManager;
+import org.springframework.cache.annotation.EnableCaching;
+import org.springframework.cache.concurrent.ConcurrentMapCacheManager;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.FilterType;
+import org.springframework.data.jpa.repository.config.EnableJpaAuditing;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import java.util.List;
+
+@ExtendWith(SpringExtension.class)
+@ContextConfiguration
+@DataJpaTest
+@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
+@EnableJpaAuditing
+public class FormSessionsPaginationTest {
```

Comment:
@snopoke I am struggling with getting this test to be configured correctly. I am getting an error `java.lang.IllegalStateException: Unable to retrieve @EnableAutoConfiguration base packages` with the current config. I don't necessarily understand all the annotations used here fully and wondering if you would be able to take a quick look and see if I am doing something obviously wrong.

Reply:
What is it you're trying to test? Can you not add tests to the existing `FormSessionServiceTest` class?

Reply:
I can though I am trying to avoid mocking `FormSessionRepo` in order to test the pagination behaviour of the repo. Though I think it would be also alright just to test our wiring aka `FormSessionService` and mock the `FormSessionRepo` pagination ?

Reply:
Up till now I've tested the two independently, the repo I've tested with actual DB access and the service has used a mock repo. I don't think the's a lot of value in testing them together but I could be convinced otherwise. 

If you do want to try set up a test with DB access I would try just copying the pattern of the `FormSessionRepoTest` and autowiring in the service. I can give that a go if you don't have any luck.

Reply:
I gave it another shot but facing the same `java.lang.IllegalStateException: Unable to retrieve @EnableAutoConfiguration base packages` issue while trying to autowire `FormSessionService` . On top of `FormSessionRepoTest` I added the `@ExtendWith(SpringExtension.class)` and `@ContextConfiguration` to indicate that this class should provide configuration and then added the config below to provide `formSessionService` - 

```
 @ComponentScan(
            basePackageClasses = {FormSessionService.class},
            useDefaultFilters = false,
            includeFilters = @ComponentScan.Filter(type = FilterType.ASSIGNABLE_TYPE, classes = {FormSessionService.class})
    )
    @EnableCaching
    @Configuration
    public static class FormSessionServiceTestConfig { }
```

Reply:
fixed in 2f54124e. You were missing the requirements for the FormSessionRepo

Reply:
Ahh I see, thanks a lot for looking at this. I think I was missing `@EnableJpaRepositories` and `@EntityScan` when I tried adding the same.